Including `wx/msw/msvcrt.h` breaks in-place `new` use

- wxWidgets version :  3.2.1 
- wxWidgets port :  wxMSW
- OS : Windows 10


i add momery leak check code as follow: 

```cpp
// ......
#include <crtdbg.h>
#include <iostream>

#ifdef __WXMSW__
#include <msvc/wx/setup.h>
// #include <wx/compiler.h>
#include <wx/msw/msvcrt.h>
#endif

inline void EnableMemLeakCheck()
{
  // _CrtSetReportMode( _CRT_ERROR, _CRTDBG_MODE_FILE );
  // _CrtSetReportFile( _CRT_ERROR, _CRTDBG_FILE_STDERR );
  _CrtSetDbgFlag(_CrtSetDbgFlag(_CRTDBG_REPORT_FLAG) | _CRTDBG_LEAK_CHECK_DF);
  std::cout<<"---------------------------- " <<__VISUALC__<<std::endl;
}

class AppMain : public wxApp
  {
  public:
    AppMain()
    {
      EnableMemLeakCheck();    
      //
      frame = nullptr;
    }

// ......
}

// ......

```

i get the error info as follow when i compile

```
[build] E:\bigdata\wxWidgets\3.2.1\include\wx/vector.h(471,15): error C2061: 语法错误: 标识符“place” 

```

--- a/include/wx/any.h
+++ b/include/wx/any.h
@@ -205,9 +205,18 @@ public:
     static void SetValue(const T& value,
                          wxAnyValueBuffer& buf)
     {
-        // Use placement new
+        // Use placement new, taking care to avoid running into problems with
+        // "new" redefinition in wx/msw/msvcrt.h.
+#ifdef WXDEBUG_NEW
+    #undef new
+#endif
+
         void* const place = buf.m_buffer;
         ::new(place) T(value);
+
+#ifdef WXDEBUG_NEW
+    #define new WXDEBUG_NEW
+#endif
     }
 
     static const T& GetValue(const wxAnyValueBuffer& buf)

--- a/include/wx/msw/msvcrt.h
+++ b/include/wx/msw/msvcrt.h
@@ -9,6 +9,16 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
 
+// Note that this file globally redefines "new" keyword breaking the use of
+// placement new in any code parsed after it. If you run into this problem, the
+// solutions are, in order of preference:
+//
+// 1. Don't include this header at all. Use better tools for memory debugging.
+// 2. Predefine __NO_VC_CRTDBG__ before including it (basically the same as 1).
+// 3. Do "#undef new" before and "#define new WXDEBUG_NEW" in your own code
+//    before/after using placement new or before/after including third part
+//    headers using it.
+
 // the goal of this file is to define wxCrtSetDbgFlag() macro which may be
 // used like this:
 //      wxCrtSetDbgFlag(_CRTDBG_LEAK_CHECK_DF);

